### PR TITLE
Merge release branch

### PR DESCRIPTION
**Pull request by mstv**
## Proposed changes

- Merge changes committed to branch `release/4.0` except
  - graph performance "hotfix"
  - translations
  - build scripts
- Resolve merge conflicts:
  * GitCommands/Settings/AppSettings.cs
  * GitUI/CommandsDialogs/FormAbout.Designer.cs
  * GitUI/CommandsDialogs/FormCommit.Designer.cs
  * GitUI/LeftPanel/RepoObjectsTree.cs
  * IntegrationTests/UI.IntegrationTests/GitCommands/GitModuleTests.cs
  * UnitTests/GitCommands.Tests/Settings/AppSettingsTests.cs
  * UnitTests/GitUI.Tests/UserControls/RevisionGrid/Graph/RevisionGraphTests.cs
 
## Screenshots 

would be many

## Test methodology 

- compared diffs v4.2-rc1 .. release/4.0 with 
- existing (merged) tests

## Please do not squash this nor create another merge commit

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/Resources/ChangeLog.md [hunk 1]
```diff
@@ -1,6 +1,237 @@
 Changelog
 =========
 
+### Version 4.2 (21 Oct 2023)
+
+#### Changes:
```
Comment (mstv) [inline GitUI/Resources/ChangeLog.md:6]: Shall we merge `ChangeLog.md`?

**Review by gerhardol — APPROVED** [review]
Thanks!
I believe the patches should go into master first in the normal case or be merged directly to master. Not many users of the release/ branches

**Comment by RussKie** [comment]
To me this is an indication 4.2 is complete, I'll get on with making the release and then we'll merge this.

**Comment by RussKie** [comment]
@mstv I took the liberty to update the branch taking in the final commits on the release/4.0 branch.
Thank you for your work.

**Pull request merged**